### DISCUSSION TITLE
Why do some highlight groups are not highlighting?

### DISCUSSION THREAD

**Discussion by Demianeen** [neovim/neovim#29281] [Q&A] [answered]
I was trying to add some sort of highlighting for the format specifier in go:
![image](https://github.com/neovim/neovim/assets/51330172/56816a5a-ad0c-4a9d-a9c5-89a520583580)
I have found that format specifiers use [`@character`](https://github.com/nvim-treesitter/nvim-treesitter/blob/9a7ad2ff7a7ea81016aca2fc89c9b2c1a5365421/queries/printf/highlights.scm#L1C11-L1C20) highlighting. However, I have an issue with highlighting this `@character` group. When I set the highlight for `@character.printf` or `@character`, nothing happens. However, setting the highlight for other syntax groups, such as `@comment`, works fine.

https://github.com/neovim/neovim/assets/51330172/a4649b45-bc20-483e-8580-bca6542e4bc5

Does anyone know what can be the issue?

**Comment by clason** [comment]
First, check that this is the correct highlight group with `:Inspect`.

**Reply by Demianeen** [reply]
Wow, thanks @clason! It doesn't have a @ character highlight group

![CleanShot 2024-06-11 at 11 57 01@2x](https://github.com/neovim/neovim/assets/51330172/f01547d4-0513-4965-9feb-5ce3eeb91de3)

Why do you think despite [mapping](https://github.com/nvim-treesitter/nvim-treesitter/blob/9a7ad2ff7a7ea81016aca2fc89c9b2c1a5365421/queries/printf/highlights.scm#L1C11-L1C20) to the @ character highlight group and the fact that treesitter sees is as format node it is still just golang string?

**Reply by Demianeen** [reply, marked as answer]
Okay, I fixed it. The query wasn't working for me, so I created my own treesitter [query](https://github.com/Demianeen/dotfiles/blob/6a6bdfc35442d8f7afcbbdc6a3aca6b7d0b67057/private_dot_config/nvim/queries/printf/highlights.scm#L2-L2) that does the same thing
